ir/memory.cpp

```diff
@@ -438,7 +438,7 @@ Memory::Memory(State &state, bool little_endian)
   // the block) should not overflow.
   avail_space = expr::mkVar("avail_space", bitsPtrSize());
   if (state.isSource()) {
-    state.addAxiom(avail_space.ult(expr::mkInt(-2, bitsPtrSize())));
+    state.addAxiom(avail_space.ult(expr::mkInt(-2 * max_align, bitsPtrSize())));
```

Comment:
The space for a byte for null pointer & a byte for the end of the memory was needed bigger than 2 due to alignment.

Reply:
sorry, didn't understand.
Why do we need to reserve more than 2 bytes? (I understand the argument for 2).

Reply:
If a big block is allocated, the next block cannot be allocated even if avail_space says it's ok.
ex)
Assume that total memory size is 8. 0x0, 0x7 is initially filled. `avail_space` is 6.
Execute `p = alloc 2, align 1`. Assume that p is allocated at `0x4`. Now 0x1, 0x2, 0x3, 0x6 are only available bytes, `avail_space` is 4.
Execute `q = alloc 1, align 4`. `avail_space >= 4`, so it says `q` can be allocated, but due to fragmentation this cannot.

Reply:
-> Sorry, I don't think this is a good example. I tried to quickly find better one, but couldn't; Decreasing -2 to something lower was needed to make tests pass. I'll leave here after finding better example.

Reply:
I see. But this change doesn't fix the problem either.
This avail_space thing needs to be revised. We need to ensure allocation only succeeds if there's address space left. Checking for avail_space is insufficient.

Reply:
Declaring `p = alloc 2, align 1` as a non-local var and `q = alloc 1, align 4` as a local one seems to be the case. `p` is not quantified, but `q` is quantified, so that was the problem.
To guarantee whether there exists a space, I think we can use `avail_space / (1 + # of blocks)` instead of `avail_space`. We can say that there exists at least such contiguous space.